### ISSUE TITLE
Add Vercel URL rewrite to allow reloading the web app

### ISSUE THREAD

**Issue by Chaphasilor** [jeffvli/feishin#168]
## Expected Behavior

https://feishin.vercel.app should be able to reload on any page within the app. Right now it only works on the homescreen, any other route results in a 404: https://feishin.vercel.app/library/songs

## Current Behavior

The problem is that React (the framework driving Feishin as a web app) is used to build a Single-Page Application (SPA). SPAs consist of one single `index.html` file that contains all of the necessary JavaScript for the app, and the script will look at the URL path to determine which content to show within the app. **No actual HTTP request is made when navigating to a different "page" within the app!** That's also why the apṗ works just fine, until you reload. Then the browser tries to load the HTML for the current URL, but there's no such file. It's all the `index.html`.

## Steps to Reproduce

1. Go to https://feishin.vercel.app
2. Click on "Tracks" in the sidebar to navigate to `/library/songs`
3. Reload the page
4. Vercel 404 is shown

![image](https://github.com/jeffvli/feishin/assets/18015147/f00f83d6-7c8a-4dce-ab16-759dc510f217)

## Possible Solution (Not obligatory)

In order to trick the browser into always loading the `index.html`, no matter the actual URL path, a *rewrite* is being used. Vercel supports these rewrites, and instead of trying to load an html file at `/library/songs`, it will *always* serve `index.html`. The necessary config can be found here: https://vercel.com/docs/frameworks/vite#using-vite-to-make-spas  
That example is for the *Vite* framework, but the same should apply to Feishin.

Basically, create a `vercel.json` file at the project root (if that doesn't exist already), and add the following (you might have to replace `/index.html` with the correct path):

```json
{
  "rewrites": [
    {
      "source": "/(.*)",
      "destination": "/index.html"
    }
  ]
}
```

## Context

It's pretty annoying to always manually edit the URL after restarting the browser, and it should be a very simple fix :)

## Your Environment

- Application version (e.g. v0.1.0) : ~~v0.0.1-alpha5~~ v0.2.0 at https://feishin.vercel.app/ (Feishin reports an old version to Jellyfin)
- Operating System and version (e.g. Windows 10) : Windows 10, Microsoft Edge Dev 116.0.1923.0
- Server and version (e.g. Navidrome v0.48.0) : Jellyfin 10.8.4
- Node version (if developing locally) :

**Comment by jeffvli** [comment]
I've tried testing with a [few different vercel rewrite configurations](https://github.com/jeffvli/feishin/commits/vercel) but I can't quite seem to get it to work. I'm out of ideas for the moment but if you or anyone else wants to try hosting on vercel and testing out some fixes that would be helpful!

**Comment by katrinabryce** [comment]
I am serving it on Nginx.

On the Nginx configuration, I have:
```
location / {
      root /usr/local/www/feishin;
      try_files $uri /index.html;
      }
    }
```
And in index.html, I changed
```
<script defer=defer src=./renderer.js></script>
<link href=./style.css rel=stylesheet>
```
to
```
<script defer=defer src=/renderer.js></script>
<link href=/style.css rel=stylesheet>
```
That worked for me

I'm not familiar with vercel, but I suspect what is happening for you is that when you visit for example
https://appurl/library/albums/
It is looking for https://appurl/library/albums/renderer.js and https://appurl/library/albums/style.css, and rewriting them as index.html.

**Comment by Chaphasilor** [comment]
Yes @katrinabryce that seems to be the issue. URLs starting with `/` are absolute paths that always start at the root, while a URL starting with `./` or without a slash is relative, and that seems to be causing issues, good catch!  
Also, I believe that there should always be quotes around attribute values in HTML to prevent strange parsing issues.  
For example:
```html
<script defer=defer src=/renderer.js></script>
<link href=/style.css rel=stylesheet>
```

should look like

```html
<script defer="defer" src="/renderer.js"></script>
<link href="/style.css" rel="stylesheet">
```

**Comment by jeffvli** [comment]
Thanks for the info!

I was able to resolve it with the `src="/renderer.js"` workaround by creating a new webpack config specifically for the web build.
The vercel build was previously using the electron renderer build which targets renderer.js using the relative path rather than the absolute.
